cmd/juju/cloud/add.go

```diff
@@ -139,8 +141,14 @@ func (c *AddCloudCommand) Run(ctxt *cmd.Context) error {
 	}
 
 	// first validate cloud input
-	if err = cloud.ValidateCloudSet([]byte(c.CloudFile)); err != nil {
-		ctxt.Warningf("%s", err.Error())
+	data, err := ioutil.ReadFile(c.CloudFile)
```

Comment:
it feels a bit wrong to be reading the file here, when ParseCloudMetadataFile above does the same thing. maybe change ParseCloudMetadataFile to return the []byte contents as well?